### ISSUE TITLE
TYPE_USE and TYPE_PARAMETER annotations should always be propagated to getters and setters

### ISSUE THREAD

**Issue by kaqqao** [rzwitserloot/lombok#1998]
For a field such as:
```java
private List<@SomeAnnotation String> strings;
```
Lombok should maintain the full type (with annotations) when generating the methods:

```java
public List<@SomeAnnotation String> getList() { ... }

public void setList(List<@SomeAnnotation String list) { ... }
```
Since these are annotations _on the type itself_ not on the field, there's no need for special listing of copyable annotations or any special handling at all, just simple unconditional propagation.

**Comment by rzwitserloot** [comment]
Yeah, except, that would take [resolution](https://github.com/rzwitserloot/lombok/wiki/LOMBOK-CONCEPT:-Resolution) so while that definitely sounds like a feature that should probably be okay to add, it's borderline impossible to actually implement.

**Comment by kaqqao** [comment]
I fully accept that I have no idea how Lombok does its thing but, to my untrained eye, copying the annotations over seems no different than copying the type parameters over.
If you can deduce the type of the field is `List<String>`, so you carry this info when generating the getter, why would deducing it as `List<@SomeAnnotation String>` be much different?

**Comment by kaqqao** [comment]
Oh, I now see you're the project owner, so you definitely know what you're talking about 😅
Feel free to close the issue.
